File: libraries/managed-exceptions-lib/src/python/managed_exceptions/argumemts/item_not_found_exception.py
from http import HTTPStatus
from managed_exceptions.error_details import ErrorDetails
from managed_exceptions.managed_exception import ManagedException

class ItemNotFoundException(ManagedException):
    def __init__(self, message: str, diagnostic_details: dict[str, str] = {}):
        super().__init__(ErrorDetails(
            status_code=HTTPStatus.NOT_FOUND,
            diagnostic_code="00404",
            diagnostic_details=diagnostic_details,
            message=message
        ))

File: services/Sphana/Libraries/Sphana.ManagedException/Arguments/ItemNotFoundException.cs
using System.Net;

namespace Sphana.ManagedException.Arguments;

/// <summary>
/// Thrown when a requested item is not found.
/// </summary>
public class ItemNotFoundException : ManagedException
{
    public ItemNotFoundException(string message, Dictionary<string, string>? diagnosticDetails = null)
        : base(HttpStatusCode.NotFound, "ITEM_NOT_FOUND", message, diagnosticDetails)
    {
    }
}